sdk/tests/tutorials/valuation/test_identifier_mismatch.py

```diff
@@ -0,0 +1,279 @@
+                transaction_type="StockIn",
+            ),
+            cls.test_data_utilities.build_transaction_request(
+                instrument_id=cls.instrument_ids[1],
+                units=100,
+                price=102,
+                currency="GBP",
+                trade_date=effective_date,
+                transaction_type="StockIn",
+            ),
+            cls.test_data_utilities.build_transaction_request(
+                instrument_id=cls.instrument_ids[2],
+                units=100,
+                price=103,
+                currency="GBP",
+                trade_date=effective_date,
+                transaction_type="StockIn",
+            ),
+        ]
+
+        cls.transaction_portfolios_api.upsert_transactions(
+            scope=TestDataUtilities.tutorials_scope,
+            code=portfolio_code,
+            transaction_request=transactions,
+        )
+
+    def upsert_quotes(self, instrument_id) -> models.UpsertQuotesResponse:
+        """
+        Upserts quotes into LUSID to be used in pricing valuation
+        :param str instrument_id: The identifier used in the upserted quotes
+        :return: UpsertQuotesResponse
+        """
+
+        prices = [
+            ("GB0008847096", 100),
+            ("GB00B1CRLC47", 200),
+            ("BMG4593F1389", 300),
+        ]
+
+        requests = [
+            models.UpsertQuoteRequest(
+                quote_id=models.QuoteId(
+                    models.QuoteSeriesId(
+                        provider=self.market_data_provider,
+                        instrument_id=price[0],
+                        instrument_id_type=instrument_id,
+                        quote_type="Price",
+                        field="mid",
+                    ),
+                    effective_at=self.effective_date,
+                ),
+                metric_value=models.MetricValue(value=price[1], unit="GBP"),
+            )
+            for price in prices
+        ]
+
+        return self.quotes_api.upsert_quotes(
+            scope=self.market_data_scope,
+            request_body={
+                "quote" + str(request_number): requests[request_number]
+                for request_number in range(len(requests))
+            },
+        )
+
+    def create_configuration_recipe(
+        self, recipe_scope, recipe_code, default_id
+    ) -> lusid.models.ConfigurationRecipe:
+        """
+        Creates a configuration recipe that can be used inline or upserted,
+        and sets the default lookup identifier for quotes
+        :param str recipe_scope: The scope for the configuration recipe
+        :param str recipe_code: The code of the the configuration recipe
+        :param str default_id: The default identifier used for looking up quotes
+        :return: ConfigurationRecipe
+        """
+
+        return models.ConfigurationRecipe(
+            scope=recipe_scope,
+            code=recipe_code,
+            market=models.MarketContext(
```

Comment:
add comments in for the `rules`, `suppliers` and `options` to explain what we are doing